### ISSUE TITLE
Where did the project go? 

### ISSUE THREAD

**Issue by davelab6** [GwenaelCasaccio/gst-visualgst#31]
https://github.com/GwenaelCasaccio/gst-visualgst/graphs/contributors shows the project stalled out... I'm curious - what happened? :)

**Comment by MrGwen** [comment]
This is a long story, but to be short the project takes too much
personal time. I was not satisfied by Gtk+ (the MacOSX and Windows
support for instance), the GLib event loop rework in the VM,
the project became too "big" for one man. For instance I've made a
browser similar to VisualGST that has 75 lines of code... So yes it was
what we can call over engineered.

And even if it was at some point really fun to do (I've learned a lot of
stuff) there were 4 users (building and IDE for such a big community is
not really motivating :D)

Cheers.

**Comment by davelab6** [comment]
Sure, makes sense. I am new to Smalltalk and I wonder why it sort of died
out.

https://www.mail-archive.com/fonc@vpri.org/msg04517.html reasonably
explains; its good for learning but became impractical in the 'real world'
:)

Thanks!

**Comment by MrGwen** [comment]
Why Smalltalk died?

It's a complex story the price of the first implementations (coupled
with the hardware), some other people args that java killed Smalltalk.
Today Smalltalk is clearly not in the hype, no linear type, no "complex"
type system (a-la Haskell, Idris, Ats, ...) Don't work on .net, java,
no monads :-), no really web friendly ...

Also what killed Smalltalk is the "real world" http, html, sql, xml,
ldap, (the list is so long...) not that is hard to do in Smalltalk
but I've the impression that Smalltalk deserves something better than
http, html, ...
Why would I use http, html when I can talk to remote objects! (without
xml, or any other annotations horrors).
Why should I use Sql when I have something like Gemstome or Magma it's
totally transparent. Only message sending!

Well I don't know what killed Smalltalk but I can tell you it's one of
the wonderful language I've ever used and still use it. Yes we could add
a type inference, a better module system (like Newspeak), get rid of the
exceptions (but keep first class context), better handling of the state
(look at the paper Worlds controlling scope of side effects) ... The
language is so much flexible that you can do it very easily!

**Comment by davelab6** [comment]
Right, I can see that for those who learn it, it is wonderful and they
become kind of hardcore about it :) But why do so few learn it? I guess
because Java and Python and C are taught in schools and people don't read
much history, or learn themselves :)
